<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
**Table of Contents**

- [awsome-rna-tools](#awsome-rna-tools)
    - [Toolbox](#toolbox)
    - [Structural databases](#structural-databases)
    - [3D annotation](#3d-annotation)
    - [List of tools (at labs)](#list-of-tools-at-labs)
        - [RNApolis](#rnapolis)
            - [RNApolis – a virtual laboratory of RNA bioinformatics](#rnapolis--a-virtual-laboratory-of-rna-bioinformatics)
            - [Computational systems for quadruplexes](#computational-systems-for-quadruplexes)
            - [Other computational tools](#other-computational-tools)
        - [bgsu.edu](#bgsuedu)

<!-- markdown-toc end -->


# awsome-rna-tools
awesome tools for RNA
https://bio.tools/

(Quick and dirty dump of tools, by Category or by Labs, to be cleaned and improved over time)

## Toolbox

- rna-tools: a toolbox to analyze sequences, structures and simulations of RNA (and more) docs @ http://rna-tools.rtfd.io web @ http://rna-tools.online https://github.com/mmagnus/rna-tools `pip install rna-tools`

## Structural databases

- http://rna.bgsu.edu/rna3dhub
- RNAsolo - database of cleaned PDB-derived RNA 3D structures https://rnasolo.cs.put.poznan.pl

## 3D annotation

- RNAspider A webserver to analyze entanglements in RNA 3D structures https://rnaspider.cs.put.poznan.pl/

## List of tools (at labs)
### RNApolis

- https://github.com/RNApolis
- https://www.cs.put.poznan.pl/mszachniuk/site/research-grants/

#### RNApolis – a virtual laboratory of RNA bioinformatics

- RNAssess	webserver for quality assessment of RNA 3D structures 
- RNAComposer	fully automated RNA structure prediction server
- RNAfitme	webserver for modeling nucleobase and nucleoside residue conformation in fixed-backbone RNA structures 
- RNA FRABASE	RNA FRAgments dataBASE and search engine
- RNAhugs*	webserver for RNA 3D structure alignment 
- RNAloops	database of RNA multiloops 
- RNAlyzer	framework for quality analysis of RNA models 
- RNApdbee	multifunctional webserver tool for RNA structure annotation
- RNAQUA	computational tool for RNA QUality Assessment 
- RNAspider	webserver to analyze entanglements in RNA 3D structures 
- RNAsolo	database of cleaned PDB-derived RNA 3D structures 
- RNAtango*	webserver for torsion-angle based similarity analysis of RNA 3D structures 
- RNAthor	webserver for the automatic normalization of RNA probing data 
- RNAtive	webserver to rank 3D RNA models and infer the native 
- RNAvista	webserver to assess RNA secondary structures with non-canonical base pairs 
- MCQ4Structures  standalone app to compute torsion angle-based similarity of molecule structures 

#### Computational systems for quadruplexes

- DrawTetrado	standalone app to draw layer diagrams representing quadruplex structures 
- ElTetrado	standalone app for identification and classification of tetrads and quadruplexes 
- ONQUADRO	database of tetrads, quadruplexes, and G4-helices 
- WebTetrado*	webserver to annotate and visualize quadruplexes in nucleic acid 3D structures 

#### Other computational tools

- SMERFA*	structural, muscular and elastic RNA finder and aligner 
- Virxicon	a lexicon of viral sequences 

### bgsu.edu
https://www.bgsu.edu/research/rna/web-applications.html

Currently we are hosting 4 web applications for analyzing RNA 3D structure:

- WebFR3D - server for finding and superimposing RNA 3D motifs
- R3D Align -  global pairwise alignment of RNA 3D structures using local superpositions
- JAR3D - predicting RNA 3D motifs in sequences
- R3D-2-MSA Server - server for accessing alignments from 3d structures
